pyathena/sqlalchemy_athena.py

```diff
@@ -31,6 +31,34 @@
 
 import pyathena
 
+BLANKS = re.compile(r"\s+")
+LIMIT_COMMENT_COLUMN = 255
+# https://docs.aws.amazon.com/athena/latest/APIReference/API_Column.html
+LIMIT_COMMENT_TABLE = 51200
+# https://docs.aws.amazon.com/athena/latest/APIReference/API_TableMetadata.html
+
+
+def process_comment_literal(value, dialect, squash_blanks=False):
+    """Escapes comments in DDL statements
+
+    String literals in COMMENT are not escaped in the same way as data
+    literal strings. Data literal use '' (double single quotes), comments escape
+    single quotes with a \\ (backslash).
+    Additionally, column comments do not support the presence of new line, carriage
+    return, etc. characters while table comments do. Hence the ``squash_blanks``
+    argument.
+    Finally, % (percent signs) need to be escaped, by being doubled, so as to not
+    be confused with placeholders for query parameters.
+    """
+    value = value.replace("\\", "\\\\").replace("'", r"\'")
+    if squash_blanks:
+        value = BLANKS.sub(" ", value)  # Replace blanks with plain spaces
```

Comment:
I don't think this process is necessary.

Reply:
Considering the problem with newlines this help mitigate the problem.

I personally see SQLAlchemy as an abstraction layer on top of multiple "SQL Engine". And for our use cases that abstraction is key. So I would like to see drivers (like PyAthena), responsible for making the abstraction become concrete, to work in a _best effort_ mode. This way, the minimum amount of information or capabilities are lost when going from abstract to concrete.

I agree that the above is not perfect. But it seems to me a decent trade-off. And it only occurs once in a table's lifetime: when it's created so the overhead of making _"our best"_ here, is really low.

Reply:
No, but I don't know of another RDBMS that supports comments but not newlines chars (escaped or not) in it. Nor do I know all RDBMS out there, so if we were to look hard enough, I am sure we'd find one. The question is more what do we want to do.

Anyhow, I removed this. I'll deal with it by other means.